File: mobile/lib/features/support_tickets/domain/support_ticket.dart
class SupportTicket {
  final String id;
  final String tenantId;
  final String? referenceNumber;
  final String subject;
  final String description;
  final String status;
  final String priority;
  final String category;
  final String submittedByUserId;
  final String? businessName;
  final String createdAt;
  final String updatedAt;
  final String? closedAt;
  final String? reopenedAt;

  const SupportTicket({
    required this.id,
    required this.tenantId,
    required this.referenceNumber,
    required this.subject,
    required this.description,
    required this.status,
    required this.priority,
    required this.category,
    required this.submittedByUserId,
    required this.businessName,
    required this.createdAt,
    required this.updatedAt,
    required this.closedAt,
    required this.reopenedAt,
  });

  factory SupportTicket.fromJson(Map<String, dynamic> json) => SupportTicket(
    id: json['id'].toString(),
    tenantId: json['tenant_id'].toString(),
    referenceNumber: json['reference_number'] as String?,
    subject: json['subject'] as String,
    description: json['description'] as String,
    status: json['status'] as String,
    priority: json['priority'] as String,
    category: json['category'] as String,
    submittedByUserId: json['submitted_by_user_id'].toString(),
    businessName: json['business_name'] as String?,
    createdAt: json['created_at'] as String,
    updatedAt: json['updated_at'] as String,
    closedAt: json['closed_at'] as String?,
    reopenedAt: json['reopened_at'] as String?,
  );

  bool get isClosed => status == 'closed';
}

File: deendoon/app/Models/SupportTicket.php
<?php

namespace App\Models;

use Database\Factories\SupportTicketFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Concerns\HasUlids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * Module 7 — Support & Tickets. Deliberately does NOT use BelongsToTenant,
 * matching ProfessionalCollectionRequest's exact precedent: this table is
 * bimodal by design — visible to its submitting tenant AND, without any
 * tenant filter at all, to the Deendoon Platform Administrator
 * (tenant_id IS NULL). Every query against this model is scoped explicitly
 * in SupportTicketController/AdminSupportTicketController/SupportTicketPolicy
 * instead.
 */
#[Fillable(['subject', 'description', 'status', 'priority', 'category', 'reference_number', 'submitted_by_user_id', 'closed_at', 'reopened_at'])]
class SupportTicket extends Model
{
    /** @use HasFactory<SupportTicketFactory> */
    use HasFactory, HasUlids;

    protected function casts(): array
    {
        return [
            'created_at' => 'datetime',
            'updated_at' => 'datetime',
            'closed_at' => 'datetime',
            'reopened_at' => 'datetime',
        ];
    }

    public function tenant(): BelongsTo
    {
        return $this->belongsTo(Tenant::class);
    }

    public function submittedBy(): BelongsTo
    {
        return $this->belongsTo(User::class, 'submitted_by_user_id');
    }

    public function messages(): HasMany
    {
        return $this->hasMany(SupportTicketMessage::class);
    }

    public function attachments(): HasMany
    {
        return $this->hasMany(SupportTicketAttachment::class);
    }
}
